server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java

```diff
@@ -45,37 +46,38 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
    public static CompletionStage<RespRequestHandler> rename(Resp3Handler handler, byte[] srcKey, byte[] dstKey,
          ChannelHandlerContext ctx, BiConsumer<?, ByteBufPool> consumer) {
       MediaType vmt = handler.cache().getValueDataConversion().getStorageMediaType();
-      final AdvancedCache<byte[], Object> acm = handler.cache()
-            .<byte[], Object>withMediaType(MediaType.APPLICATION_OCTET_STREAM, vmt);
-      return acm.removeAsyncEntry(srcKey).thenCompose(e -> {
-         if (e == null) {
-            RespErrorUtil.noSuchKey(handler.allocator());
-            return handler.myStage();
-         }
-         CompletableFuture<Long> myStage;
-         if (Arrays.equals(srcKey, dstKey)) {
-            // If src = dest ...
-            myStage = CompletableFuture.completedFuture(1L);
-         } else {
-            var timeService = handler.respServer().getTimeService();
-            // ... else if Immortal entry case: copy metadata...
-            if (e.getLifespan() <= 0) {
-               var newMeta = e.getMetadata().builder();
-               myStage = acm.putAsyncEntry(dstKey, e.getValue(), newMeta.build()).thenApply(ignore -> 1L);
-            } else {
-               long newLifespan = e.getLifespan() + e.getCreated() - timeService.wallClockTime();
-               if (newLifespan > 0) {
-                  // ... else if not expired copy metadata and preserve lifespan...
-                  var newMeta = e.getMetadata().builder().lifespan(newLifespan);
-                  myStage = acm.putAsyncEntry(dstKey, e.getValue(), newMeta.build()).thenApply(ignore -> 1L);
+      final AdvancedCache<byte[], Object> acm = handler.typedCache(vmt);
+      CompletionStage<Long> cs = acm.removeAsyncEntry(srcKey)
+            .thenCompose(e -> {
+               if (e == null) {
+                  RespErrorUtil.noSuchKey(handler.allocator());
```

Comment:
I remembered that the allocator is also not thread safe and should not be used in a lambda like this.

We should probably add `assert ctx.channel().eventLoop().inEventLoop();` to the `RespHandler#allocateBuffer` method as well.

Reply:
I'll start with that and work backwards trying to make everything safe.